crates/oxc_linter/src/rules/deepscan/bad_array_method_on_arguments.rs

```diff
@@ -0,0 +1,202 @@
+        ("function fn() {arguments.toLocaleString(() => {})}", None),
+        ("function fn() {arguments.toString(() => {})}", None),
+        // keep pass for DeepScan compatible
+        ("function fn() {arguments.findLast(() => {})}", None),
+        ("function fn() {arguments.group(() => {})}", None),
+        ("function fn() {arguments.groupToMap(() => {})}", None),
+        ("function fn() {arguments.toReversed(() => {})}", None),
+        ("function fn() {arguments.toSorted(() => {})}", None),
+        ("function fn() {arguments.toSpliced(0)}", None),
+        ("function fn() {arguments.with(1, 1)}", None),
+    ];
+
+    let fail = vec![
+        ("function fn() {arguments['map'](() => {})}", None),
+        ("function fn() {arguments[`map`](() => {})}", None),
+        ("function fn() {arguments.at(0)}", None),
+        ("function fn() {arguments.concat([])}", None),
+        ("function fn() {arguments.copyWithin(0)}", None),
+        ("function fn() {arguments.entries()}", None),
+        ("function fn() {arguments.every(() => {})}", None),
+        ("function fn() {arguments.fill(() => {})}", None),
+        ("function fn() {arguments.filter(() => {})}", None),
+        ("function fn() {arguments.find(() => {})}", None),
+        ("function fn() {arguments.findIndex(() => {})}", None),
+        ("function fn() {arguments.flat(() => {})}", None),
+        ("function fn() {arguments.flatMap(() => {})}", None),
+        ("function fn() {arguments.forEach(() => {})}", None),
+        ("function fn() {arguments.includes(() => {})}", None),
+        ("function fn() {arguments.indexOf(() => {})}", None),
+        ("function fn() {arguments.join()}", None),
+        ("function fn() {arguments.keys()}", None),
+        ("function fn() {arguments.lastIndexOf('')}", None),
+        ("function fn() {arguments.map(() => {})}", None),
+        ("function fn() {arguments.pop()}", None),
+        ("function fn() {arguments.push('')}", None),
+        ("function fn() {arguments.reduce(() => {})}", None),
+        ("function fn() {arguments.reduceRight(() => {})}", None),
+        ("function fn() {arguments.reverse()}", None),
+        ("function fn() {arguments.shift()}", None),
+        ("function fn() {arguments.slice()}", None),
+        ("function fn() {arguments.some(() => {})}", None),
+        ("function fn() {arguments.sort(() => {})}", None),
+        ("function fn() {arguments.splice(() => {})}", None),
+        ("function fn() {arguments.unshift()}", None),
+        ("function fn() {arguments.values()}", None),
+        ("function fn() {arguments['@@iterator'](() => {})}", None),
+    ];
+
+    Tester::new(BadArrayMethodOnArguments::NAME, pass, fail).test_and_snapshot();
+}
+
+
+#[test]
+fn test_array_is_sorted() {
+    let is_sorted = |arr: &[&str]| -> bool {
+        for i in 1..arr.len() {
+            if arr[i] < arr[i - 1] {
+                println!("{}, {}", arr[i], arr[i-1]);
```

Comment:
you probably meant assert?

Reply:
![image](https://user-images.githubusercontent.com/1430279/233682576-23841265-34e6-447d-b993-e5535964c027.png)

Haha, this is from the code coverage, the red shows it's unreached, who's the correct one here 😉 ?

Reply:
FYI you can just sort the array and compare it with the original array